src/platforms/mesa/server/kms/real_kms_output_container.cpp

```diff
@@ -41,12 +41,23 @@ void mgm::RealKMSOutputContainer::update_from_hardware_state()
 {
     decltype(outputs) new_outputs;
 
+    // TODO: Accumulate errors and present them all.
+    std::exception_ptr last_error;
+
     for (auto drm_fd : drm_fds)
     {
-        kms::DRMModeResources resources{drm_fd};
-
+        std::unique_ptr<kms::DRMModeResources> resources;
+        try
+        {
+            resources = std::make_unique<kms::DRMModeResources>(drm_fd);
+        }
+        catch (std::exception const&)
+        {
+            last_error = std::current_exception();
```

Comment:
Do we need to "accumulate" errors? Can't we just log failures? Vis:
```c++
            log(Severity::informational, MIR_LOG_COMPONENT, last_error, "Node does not support KMS");
```

Reply:
This is called relatively frequently, so I think it would be unnecessarily noisy to log each time? Maybe?